agent/rename_directories.go

```diff
@@ -12,12 +15,26 @@ import (
 func (s *Server) RenameDirectories(ctx context.Context, in *idl.RenameDirectoriesRequest) (*idl.RenameDirectoriesReply, error) {
 	gplog.Info("agent received request to rename segment data directories")
 
+	mErr := &multierror.Error{}
+
 	for _, pair := range in.GetPairs() {
-		err := utils.System.Rename(pair.Src, pair.Dst)
-		if err != nil {
-			return &idl.RenameDirectoriesReply{}, err
+
+		if err := PostgresOrNonExistent(pair.Src); err != nil {
+			mErr = multierror.Append(mErr, err)
+			continue
+		}
```

Comment:
I don't understand what the goal of this helper function is.

Reply:
For agent functions that delete or rename data directories, I wanted to make sure there are at least some sanity checks that that directory is a postgres data directory.  Hence the checks for the postgres files `postgresql.conf` and `PG_VERSION`.  However, if a repeat call is made (due to a re-run of this substep for idempotence), those files might not exist because the dataDir itself has already been moved/removed.  

Perhaps a better name would help?  I tried a bit for one.